File: 09-missing-number/MissingNumber.java
public class MissingNumber {
    public static int missingNumber(int[] nums) {
        int num = nums.length;

        for (int i = 0; i < nums.length; i++) {
            if (num == nums[i]) num--;
            return num;
        }
        return num;
    }
}

File: 09-missing-number/MissingNumber.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n=0, i=0;
        for(int j=0;j<nums.size();j++){
            n^=j+1;
            i^=nums[j];
        }

        return n^i;
    }
};